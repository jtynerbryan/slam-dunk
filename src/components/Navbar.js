import React, { Component } from 'react'
import { Menu } from 'semantic-ui-react'

class Navbar extends Component {
  state = {}

  handleItemClick = (e, { name }) => this.setState({ activeItem: name })

  render() {
    const { activeItem } = this.state

    return (
      <Menu inverted>
        <Menu.Item
          position='right'
          name='GitHub'
          active={activeItem === 'github'}
          onClick={this.handleItemClick}
          href='http://github.com/jtynerbryan' target="_blank"
        >
          GitHub
        </Menu.Item>
      </Menu>
    )
  }
}

export default Navbar
