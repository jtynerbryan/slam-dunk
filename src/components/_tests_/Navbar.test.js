import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

describe('Navbar', () => {
  it('renders without crashing', () => {
    shallow(<Navbar />);
  });

  it('renders without unexpected changes', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
