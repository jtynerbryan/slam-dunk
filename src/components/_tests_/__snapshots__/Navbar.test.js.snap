// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders without unexpected changes 1`] = `
<div
  className="ui inverted menu"
>
  <a
    className="right item"
    href="http://github.com/jtynerbryan"
    onClick={[Function]}
    target="_blank"
  >
    GitHub
  </a>
</div>
`;
